src/credentials/CHIPOpCred.h

```diff
@@ -0,0 +1,184 @@
+     **/
+    const CertificateKeyId * GetTrustedRootId(uint16_t certSetIndex) const;
+
+    /**
+     * @brief Check whether certificate set is in the operational credential set.
+     *
+     * @param cert  Pointer to the ChipCertificateSet structures.
+     *
+     * @return True if certificate is in the set, false otherwise.
+     **/
+    bool IsCertSetInTheOpCredSet(const ChipCertificateSet * cert) const;
+
+    /**
+     * @brief Check whether Trusted Root ID is in the operational credential set.
+     *
+     * @param trustedRoot  Reference to the Trusted Root ID.
+     *
+     * @return True if Trusted Root ID is in the set, false otherwise.
+     **/
+    bool IsTrustedRootIn(const CertificateKeyId & trustedRoot) const;
+
+    /**
+     * @brief Validate CHIP certificate.
+     *
+     * @param trustedRootId    Reference to the Trusted Root ID for the Certificate Set to be used
+     *                          for validation
+     * @param cert              Pointer to the CHIP certificiate to be validated. The certificate is
+     *                          required to be in this set, otherwise this function returns error.
+     * @param context           Certificate validation context.
+     *
+     * @return Returns a CHIP_ERROR on validation or other error, CHIP_NO_ERROR otherwise
+     **/
+    CHIP_ERROR ValidateCert(const CertificateKeyId & trustedRootId, const ChipCertificateData * cert, ValidationContext & context);
+
+    /**
+     * @brief Find and validate CHIP certificate.
+     *
+     * @param trustedRootId    Reference to the Trusted Root ID for the Certificate Set to be used
+     *                          for validation
+     * @param subjectDN         Subject distinguished name to use as certificate search parameter.
+     * @param subjectKeyId      Subject key identifier to use as certificate search parameter.
+     * @param context           Certificate validation context.
+     * @param cert              A pointer to the valid CHIP certificate that matches search criteria.
+     *
+     * @return Returns a CHIP_ERROR on validation or other error, CHIP_NO_ERROR otherwise
+     **/
+    CHIP_ERROR FindValidCert(const CertificateKeyId & trustedRootId, const ChipDN & subjectDN,
+                             const CertificateKeyId & subjectKeyId, ValidationContext & context, ChipCertificateData *& cert);
+
+    /**
+     * @brief A function to sign a msg using ECDSA and the respective device credentials keypair.
+     *
+     * @param msg           Message that needs to be signed
+     * @param msg_length    Length of message
+     * @param out_signature Buffer that will hold the output signature. The signature consists of: 2 EC elements (r and s),
+     *                      represented as ASN.1 DER integers, plus the ASN.1 sequence Header
+     *
+     * @return Returns a CHIP_ERROR on error, CHIP_NO_ERROR otherwise
```

Comment:
Please delete obvious doxygens. The take up vertical space when reading the header and do not seem to help in understanding code.